polaris_bitgo/bitgo/bitgo.py

```diff
@@ -126,12 +129,17 @@ def get_public_key(self) -> str:
         """
         return self.wallet.public_key
 
-    def get_private_key(self) -> str:
+    def get_stellar_transaction_id(self, transaction_id: str) -> str:
         """
-        Returns the private key from a signer, in this case, the user key.
-        It doesn't return the private key from the account that has the
-        supplies.
+        Gets the Stellar Network's transaction id.
 
-        :return: Returns the signer private key hash.
+        :param: The BitGo's transfer id.
+        :return: Returns a string containing the Stellar Network transaction id.
         """
-        return self._decrypt_private_key()
+        while True:
+            response_data = self.bitgo_api.get_transfer_by_id(transaction_id)
+            if response_data.get("state") == FAILED_STATUS:
+                raise RuntimeError("BitGo failed to complete the transfer.")
+            if response_data.get("state") == CONFIRMED_STATUS:
+                break
+        return response_data.get("txid")
```

Comment:
What do you think about something like this to avoid the `while True` approach? It's just a personal thought that while loops should have their "break logic" on their declarations and avoid `breaks` statements alongside the code. But it's just a suggestion :D

```suggestion
        response_data = self.bitgo_api.get_transfer_by_id(transaction_id)
        while response_data.get("state") != CONFIRMED_STATUS:
            if response_data.get("state") == FAILED_STATUS:
                raise RuntimeError("BitGo failed to complete the transfer.")
            response_data = self.bitgo_api.get_transfer_by_id(transaction_id)
        return response_data.get("txid")
```

Reply:
I think this would not be worth it considering all future changes to line 139 will demand changes to 143, which is something that I believe to be easily forgettable. I found a good reference indicating that "while True + break" is appropriate:
["Users of the language are advised to use the while-True form with an inner if-break when a do-while loop would have been appropriate."](https://www.python.org/dev/peps/pep-0315/#id6)

Reply:
Sure, makes sense. I think a "do while" would be perfect. That is fine, thanks for explaining!